sale_coupon_order_line_link/models/sale_order.py

```diff
@@ -115,28 +98,20 @@ def _link_reward_generated_lines(self, programs):
         """Link the lines that generated reward lines to those lines"""
         for program in programs:
             reward_lines = self.order_line.filtered(
-                lambda x: x.coupon_program_id == program
+                lambda x: x.loyalty_program_id == program
             )
             self.order_line._filter_related_program_lines(program).write(
                 {"reward_generated_line_ids": [(4, rl.id) for rl in reward_lines]}
             )
 
-    def _create_new_no_code_promo_reward_lines(self):
-        """Ensure that the links remain"""
-        res = super()._create_new_no_code_promo_reward_lines()
-        for order in self:
-            order._link_reward_generated_lines(order.order_line.coupon_program_id)
-        return res
-
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
-    coupon_program_id = fields.Many2one(
-        comodel_name="coupon.program",
-        ondelete="restrict",
-        string="Coupon Program",
+    loyalty_program_id = fields.Many2one(
+        related="reward_id", store=True, string="Loyalty Program"
```

Comment:
```suggestion
        related="reward_id.program_id", store=True, string="Loyalty Program"
```

Reply:
Corrected the relation and the view fields.